x-pack/plugins/task_manager/server/lib/get_next_run_at.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { intervalFromDate } from './intervals';
+import type { ConcreteTaskInstance } from '../task';
+
+export function getNextRunAt(
+  task: ConcreteTaskInstance,
+  taskUpdates: Partial<Pick<ConcreteTaskInstance, 'runAt' | 'schedule'>>,
+  taskDelayThresholdForPreciseScheduling: number = 0
+): Date {
+  const { runAt: originalRunAt, startedAt, schedule } = task;
+  const { runAt: newRunAt, schedule: newSchedule } = taskUpdates;
+
+  if (newRunAt) {
+    return newRunAt;
+  }
+
+  const taskSchedule = newSchedule?.interval ?? schedule?.interval;
+  const taskDelay = Date.now() - originalRunAt.getTime();
```

Comment:
should this use `startedAt` instead of `Date.now()`?

Reply:
Good catch, otherwise it would only work if the task finished running within the poll interval 🙈 fixed in https://github.com/elastic/kibana/pull/190093/commits/3d5eb391ffcd78675107fc771f012c79b2fc3c45.